module/zfs/dmu_object.c

```diff
@@ -31,6 +31,8 @@
 #include <sys/zap.h>
 #include <sys/zfeature.h>
 
+int zfs_metadnode_backfill = 0;
```

Comment:
Yes, if we can reliably detect holes using openzfs/openzfs#82, and add appropriate handling of `ESRCH` from `dmu_object_next()`, then this patch shouldn't be needed.